### PR TITLE
Client encryption: Adds validation that partition key paths are not encrypted

### PR DESCRIPTION
**Pull request by abhijitpai**
## Description

Validate partition key paths (i.e. their top level ancestors) are not specified for encryption in the client encryption policy.

Ensure that policies with newer versions can de-serialize correctly, and expose the version so that the encryption package can fail correctly upon newer policy versions.

## Type of change
- [] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml [hunk 1]
```diff
@@ -1436,7 +1436,7 @@
             │       └── Change Feed Transport(00000000-0000-0000-0000-000000000000)  Transport-Component  MemberName@FilePath:42  12:00:00:000  0.00 milliseconds  
             └── MoveNextAsync(00000000-0000-0000-0000-000000000000)  Pagination-Component  MemberName@FilePath:42  12:00:00:000  0.00 milliseconds  
                 └── [9F-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF,BF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF-FF) move next(00000000-0000-0000-0000-000000000000)  Pagination-Component  MemberName@FilePath:42  12:00:00:000  0.00 milliseconds  
-                    └── Change Feed Transport(00000000-0000-0000-0000-000000000000)  Transport-Component  MemberName@FilePath:42  12:00:00:000  0.00 milliseconds 
+                    └── Change Feed Transport(00000000-0000-0000-0000-000000000000)  Transport-Component  MemberName@FilePath:42  12:00:00:000  0.00 milliseconds
```
Comment (kr-santosh) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml:1439]: i think this was introduced recently,

**Review by anujtoshniwal — APPROVED** [review]

**Review by kr-santosh — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs [hunk 2]
```diff
@@ -690,6 +690,11 @@ internal void ValidateRequiredProperties()
             {
                 this.indexingPolicyInternal.IncludedPaths.Add(new IncludedPath() { Path = IndexingPolicy.DefaultPath });
             }
+
+            if (this.ClientEncryptionPolicy != null)
+            {
+                this.ClientEncryptionPolicy.ValidatePartitionKeyPathsAreNotEncrypted(this.PartitionKeyPathTokens);
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs:696]: Will this be called once user create ContainerProperties and then set ClientEncryptionPolicy ?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml:1439]: Please revert the white space change

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs (hunk 2 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs:696]: Nevermind I see now, we are calling ValidateRequiredProperties on container create call

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml (hunk 1 above)
Comment (abhijitpai) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.ScenariosAsync.xml]: It was auto-generated, but have reverted it.

**Review by anujtoshniwal — APPROVED** [review]

**Review by j82w — APPROVED** [review]

**Pull request merged**